packages/core/src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -479,16 +484,28 @@ export async function startTransformationJob(uploadId: string) {
         getLogger().info(`CodeTransformation: jobId: ${jobId}`)
     } catch (error) {
         getLogger().error(`CodeTransformation: ${CodeWhispererConstants.failedToStartJobNotification}`, error)
-        if ((error as Error).message.includes('too many active running jobs')) {
+        const errorMessage = (error as Error).message
+        if (errorMessage.includes('too many active running jobs')) {
```

Comment:
It would be nice if we could centralize all of this error handling (setting the notification / chat messages) in one class, as right now this functionality gets scattered across several different files / classes

Reply:
Yeah agreed, from UX we soon will have to modify this so that error messages only show in notification if the chat is not visible to user, when that work is done I think will be a good time to refactor how we set the error messages.

Reply:
there are some useful patterns (and util functions) in `packages/core/src/shared/errors.ts`